sled-hardware/src/lib.rs

```diff
@@ -74,7 +74,16 @@ pub enum SledMode {
 
 /// Describes properties that should uniquely identify a Gimlet.
 #[derive(
-    Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize, JsonSchema,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Serialize,
+    Deserialize,
+    JsonSchema,
 )]
 #[serde(tag = "type", rename_all = "snake_case")]
 pub enum Baseboard {
```

Comment:
`Baseboard` is a confusing name to me; is this equivalent to a serial number?  It would be good to add comments tying the `Gimlet` parameters back to values defined in RFD 218 / 308.

Reply:
I believe there was a long discussion on this name. @zephraph can probably provide details. I'm not sure this change is relevant to this PR. I'm specifically using `Baseboard` as the ID here because we know it's unique per sled and is also what RSS and wicket use to map specific sleds to bootstrap IPs.